cmd/mayactl/app/command/volume_info.go

```diff
@@ -0,0 +1,159 @@
+		"unique volume name.")
+	return cmd
+}
+
+// Validate verifies the command whether volName is passed or not.
+func (c *CmdVolumeInfoOptions) Validate(cmd *cobra.Command) error {
+	if c.volName == "" {
+		return errors.New("--volname is missing. Please try running [mayactl volume list] to see list of volumes.")
+	}
+	return nil
+}
+
+// TODO : Add more volume information
+// RunVolumeInfo runs info command and display the outputs in standard I/O.
+// Currently It displays volume access modes and target portal details only.
+func (c *CmdVolumeInfoOptions) RunVolumeInfo(cmd *cobra.Command) error {
+	var (
+		address, mode []string
+		replicaCount  int
+		portalInfo    PortalInfo
+	)
+	const (
+		replicaTemplate = `
+================= Replica Details =================
+IP            AccessMode               Status
+{{range $key, $value := .}}
+{{$value.IP}}     {{$value.AccessMode}}                       {{$value.Status}}
+{{end}}
+===================================================
+`
+		portalTemplate = `
+================= Portal Details ==================
+IQN     :   {{.IQN}}
+Volume  :   {{.VolumeName}}
+Portal  :   {{.Portal}}
+Size    :   {{.Size}}
+Status  :   {{.Status}}
+===================================================
+`
+	)
+	annotation := &Annotations{}
+	// GetVolumeAnnotation is called to get the volume controller's info such as
+	// controller's IP, status, iqn, replica IPs etc.
+	err := annotation.GetVolAnnotations(c.volName)
+	if err != nil {
+		return nil
+	}
+	if annotation == nil {
+		fmt.Println("annotation is nil")
+		return nil
+	}
+	if annotation.ControllerStatus != "Running" {
+		fmt.Println("Volume not reachable, ", annotation.ControllerStatus)
+		return nil
+	}
+	// Initiallize an instance of ReplicaCollection, json response recieved from the
+	// controllerIP:9501/v1/replicas is to be parsed into this structure via GetVolumeStats.
+	// An API needs to be passed as argument.
+	collection := client.ReplicaCollection{}
+	controllerClient := client.ControllerClient{}
+	_, err = controllerClient.GetVolumeStats(annotation.ClusterIP+v1.ControllerPort, v1.InfoAPI, &collection)
+	if err != nil {
+		return err
+	}
+	portalInfo = PortalInfo{
+		annotation.Iqn,
+		c.volName,
+		annotation.TargetPortal,
+		annotation.VolSize,
+		annotation.ControllerStatus,
+	}
+	tmpl, err1 := template.New("VolumeInfo").Parse(portalTemplate)
+	if err1 != nil {
+		fmt.Println("Can't Parse the template, found error :", err1)
+		return nil
+	}
+	err = tmpl.Execute(os.Stdout, portalInfo)
+	if err != nil {
+		fmt.Println("Can't execute the template, found error :", err)
+		return nil
+	}
+	replicaInfo := make(map[int]*ReplicaInfo)
+	for key, _ := range collection.Data {
+		address = append(address, strings.TrimSuffix(strings.TrimPrefix(collection.Data[key].Address, "tcp://"), ":9502"))
+		mode = append(mode, collection.Data[key].Mode)
+		replicaInfo[key] = &ReplicaInfo{address[key], mode[key], "Running"}
```

Comment:
How does this handle the case where replica pod is non-running state?

Reply:
@kmova that will never happen, collection.Data only returns the status of running replicas

Reply:
I have handled such case below.Although it's hardcoded due to the limitation of the response we are getting.